File: CaesarCipher/caesarcipher.cpp
#include <iostream>
#include <string>
#include <ctype.h>

using namespace std;

int main()
{
    int n;
    //cout <<"Enter the length of string : ";
    cin >> n;

    string s;
    //cout <<"Enter the string to encrypt : ";
    cin >> s;

    int k;
    //cout <<"Enter the number of rotations : ";
    cin >> k;
    k = k % 26;
    for(char c : s)
    {
        if( isalpha(c) == 0 )
            cout << c;
        else    //If it is an alphabet
        {
            unsigned char en = c + k;
            if( isupper(c) && en > 90 )
                en = en - 26;
            else if(islower(c) && en > 122)
                en = en - 26;
            cout <<en ;
        }
    }

    return 0;
}

File: CaesarCipher/caesarcipher.py
n = int(input())
s = input()
k = int(input())
k = k % 26;
for c in s:
    ascii = ord(c)
    newchar = c
    if(c.isalpha()):
        newchar = ascii + k
        if    c.isupper() and newchar > 90:    
            newchar -= 26
        elif c.islower() and newchar > 122:
            newchar -= 26            
        newchar = chr(newchar)
    #print(c , ascii , newchar)    
    print(newchar, end='')
